### ISSUE TITLE
"... is only available in iOS 13.0 or newer" in a project targeting iOS 15.0

### ISSUE THREAD

**Issue by jamesncl** [freyaalminde/watch-date-picker#1]
I'm using Xcode 13.4 and have created a new SwiftUI project using the "iOS app with watch app" project template. The target for the main app is set to iOS 15.5 and the watch target is set to WatchOS 8. Everything was building and running on the simulator fine.

I've just added the [watch-date-picker Swift package](https://swiftpackageindex.com/freyaariel/watch-date-picker) to my project, and now I can't build my project - I get tons of errors in the added watch-date-picker package's code complaining that various things are only available in iOS 13.0 or newer. I haven't even tried using the library yet - I've just added it via File -> Add Packages

I'm confused why I'm getting this error - as far as I can tell, I'm targeting the latest versions of everything - iOS 15.5 and WatchOS 8.5. Can anyone explain why this is and what I can do to resolve it? Am I missing something? Is there something I need to do to tell the package to only build for WatchOS and not iOS? I've tried cleaning, deleting derived data, closing Xcode, nothing works. Thanks!
<img width="464" alt="Screenshot 2022-06-04 at 20 44 15" src="https://user-images.githubusercontent.com/15979056/172047453-781dd86b-a9ce-4bee-8bfb-ff51aafae08b.png">
<img width="468" alt="Screenshot 2022-06-04 at 20 46 11" src="https://user-images.githubusercontent.com/15979056/172047457-6522e32a-b439-484c-b162-634d3bc03e03.png">
<img width="721" alt="Screenshot 2022-06-04 at 20 50 41" src="https://user-images.githubusercontent.com/15979056/172047461-1633db3f-b368-4ddb-815c-80779b8827a9.png">

**Comment by freyaariel** [comment]
I’ve just tried it, and it turns out you have to add the library to the WatchKit extension target, not the main target, otherwise the 128 buildtime errors happen.

When adding the package, you have to select “YourApp WatchKit Extension” under “Add to Target” in this step:

![Screen Shot 2022-06-06 at 11 53 48](https://user-images.githubusercontent.com/72786/172139817-70f4120f-dc40-4c96-b607-121152827d00.png)

Then it will show up here:

![Screen Shot 2022-06-06 at 11 56 24](https://user-images.githubusercontent.com/72786/172139846-c52e2923-0164-46da-a7d4-36e9edeb9ed5.png)

Thank you for bringing this up. I’ll try to make the README’s installation instructions clearer.

**Comment by jamesncl** [comment]
Thanks for the reply!

That's interesting. What version of Xcode are you using and how are you adding the package? For me, I'm using Xcode 13.4, and I'm using `File -> Add packages...` and this is the window I get - under "Add to project" I only get 1 option, to add to the main project. There is no option to add to the watch extension:

![Screenshot 2022-06-07 at 17 52 16](https://user-images.githubusercontent.com/15979056/172439020-051217f2-624f-4d21-a343-708611aab635.png)

**Comment by freyaalminde** [comment]
Sorry for the late reply. Is this still an issue?

**Comment by jamesncl** [comment]
Thank you! I worked around the issue by manually copying the library into my project. Just tried again and this time I got the option to add to my watch extension - not sure why it didn't appear before. Now it works!

Also, thank you for the continuing updates - I was getting an error `CALayer position contains NaN` on certain models of Apple Watch on the earlier version I was using, which now seems to be fixed. Awesome!
